mozetl/taar/taar_similarity.py

```diff
@@ -11,61 +11,66 @@
 logger = logging.getLogger(__name__)
 RAND_SEED = 42
+KDE_KERNEL_BANDWIDTH = 0.35
 
 
 def get_samples(spark):
     """ Get a dataframe with a valid set of sample to base the next
     processing on.
     """
-    return spark.sql("SELECT * FROM longitudinal")\
-        .where("active_addons IS NOT null")\
-        .where("size(active_addons[0]) > 2")\
-        .where("size(active_addons[0]) < 100")\
-        .where("normalized_channel = 'release'")\
+    return spark.sql("SELECT * FROM longitudinal") \
+        .where("active_addons IS NOT null") \
+        .where("size(active_addons[0]) > 2") \
+        .where("size(active_addons[0]) < 100") \
+        .where("normalized_channel = 'release'") \
         .selectExpr(
-            "client_id as client_id",
-            "active_addons[0] as active_addons",
-            "geo_city[0] as geo_city",
-            "subsession_length[0] as subsession_length",
-            "settings[0].locale as locale",
-            "os as os",
-            "places_bookmarks_count[0].sum AS bookmark_count",
-            "scalar_parent_browser_engagement_tab_open_event_count[0].value AS tab_open_count",
-            "scalar_parent_browser_engagement_total_uri_count[0].value AS total_uri",
-            "scalar_parent_browser_engagement_unique_domains_count[0].value AS unique_tlds"
-            )
+        "client_id as client_id",
+        "active_addons[0] as active_addons",
+        "geo_city[0] as geo_city",
+        "subsession_length[0] as subsession_length",
+        "settings[0].locale as locale",
+        "os as os",
+        "places_bookmarks_count[0].sum AS bookmark_count",
+        "scalar_parent_browser_engagement_tab_open_event_count[0].value AS tab_open_count",
+        "scalar_parent_browser_engagement_total_uri_count[0].value AS total_uri",
+        "scalar_parent_browser_engagement_unique_domains_count[0].value AS unique_tlds"
+    )
 
 
 def get_addons_per_client(users_df, minimum_addons_count):
     """ Extracts a dataframe that contains one row
     for each client along with the list of active addon guids.
     """
+
     def is_valid_addon(addon):
-        if addon.is_system or\
-           addon.app_disabled or\
-           addon.type != "extension" or\
-           addon.user_disabled or\
-           addon.foreign_install:
+        if addon.is_system or \
+                addon.app_disabled or \
+                addon.type != "extension" or \
+                addon.user_disabled or \
+                addon.foreign_install:
             return False
 
         return True
 
     # TODO: Better ways to do this filtering?
-    return users_df.rdd\
+    return users_df.rdd \
         .map(lambda p: (p["client_id"],
-             [guid for guid, data in p["active_addons"].items() if is_valid_addon(data)]))\
-        .filter(lambda p: len(p[1]) > minimum_addons_count)\
-        .toDF()\
+                        [guid for guid, data in p["active_addons"].items() if is_valid_addon(data)])) \
```

Comment:
Here too! And in other places below. Rule of thumb: if it's not vital to your PR, it shouldn't go in your PR :) Otherwise it's difficult to rebase and review.